server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -262,4 +329,50 @@ private static <V> V removeMatch(Map<Set<RelationName>, V> valuesByNames, Set<Re
         }
         return null;
     }
+
+    private static final CollectOuterColumnVisitor COLLECT_OUTER_COLUMN_VISITOR = new CollectOuterColumnVisitor();
```

Comment:
Is it possible to reuse something from Symbols. or SymbolVisitors for this? 

Part of this seems fairly generic in that it could potentially live outside of the JoinPlanBuilder and closer to other symbol traversal components

Reply:
I will look into this.